### ISSUE TITLE
entities do not reinitialize when detached and then reattached to DOM

### ISSUE THREAD

**Issue by jarstin07** [aframevr/aframe#1444]
**Description:**

Able to use **detach** to remove an a-entity, but **append** does not work to place it back on the scene.

See here, where .redCube is an a-entity that is a cube, and masterScene is the class applied to a-scene.

```
$( ".redCube" ).click(function() {
  $( this ).toggleClass( "off" );
});
var p;
$( "button" ).click(function() {
  if ( p ) {
    p.appendTo( ".masterScene" );
    p = null;
  } else {
    p = $( ".redCube" ).detach();
  }
});
```

```
  <a-assets>
  <a-mixin id="cube" geometry="primitive: box"></a-mixin>
  <a-mixin id="cube-hovered" material="color: magenta"></a-mixin>
  <a-mixin id="cube-selected" material="color: cyan"></a-mixin>
  <a-mixin id="red" material="color: red"></a-mixin>
  <a-mixin id="green" material="color: green" data></a-mixin>
  </a-assets>

 <a-entity position="0 2.2 4">
    <a-entity camera look-controls wasd-controls>
      <a-entity position="0 0 -3"
                geometry="primitive: ring; radiusOuter: 0.10;
                          radiusInner: 0.08;"
                material="color: cyan; shader: flat"
                cursor="maxDistance: 30; fuse: true">
        <a-animation begin="click" easing="ease-in" attribute="scale"
             fill="backwards" from="0.1 0.1 0.1" to="1 1 1" dur="150"></a-animation>
        <a-animation begin="fusing" easing="ease-in" attribute="scale"
             fill="forwards" from="1 1 1" to="0.1 0.1 0.1" dur="1500"></a-animation>
      </a-entity>
    </a-entity>
  </a-entity>

<a-scene class="masterScene" >
      <a-entity position="0 0 -6" class="greenCube mixin="cube green">
                   <a-animation begin="click" attribute="rotation" to="0 360 0"
                   easing="linear" dur="2000" fill="backwards"></a-animation>
     </a-entity>
      <a-entity position="5 2 -6" class="redCube" mixin="cube red">
                   <a-animation begin="click" attribute="rotation" to="0 360 0"
                   easing="linear" dur="2000" fill="backwards"></a-animation>
     </a-entity>
</a-scene>
```

- A-Frame Version: 0.2.0
- Platform/Device: Mac/Chrome and tested on iphone6+/ios 9.3.1

**Comment by ngokevin** [comment]
Is this just a jQuery thing, or maybe a bug in your code snippet?

**Comment by donmccurdy** [comment]
I can reproduce this here: https://sandbox.donmccurdy.com/vr/walls/

For either one of the `<a-box/>` elements:

```javascript
sceneEl.remove(boxEl);
sceneEl.appendChild(boxEl);             // doesn't work
sceneEl.appendChild(boxEl.cloneNode()); // works
```

**Comment by dmarcos** [comment]
It does not work after remove you mean? We call remove in the components when removed from the DOM

**Comment by donmccurdy** [comment]
Seems fine that it stops working *while* it's removed, what's a bit unexpected is that it doesn't re-initialize when it is reattached, unless you clone the node first.

If there are issues preventing an entity from being re-attached, once `remove()` has been called on its components, maybe we should throw an error if we can detect a recycled entity being inserted?
